### PR TITLE
3796 fix dependencies modal theme

### LINKED ISSUE #3796: Poor text contrast in the "Add dependencies" modal when using light themes
# 🐛 bug report

## Description of the problem

The text is in white inside the "Add dependencies" modal when a light theme is used, as shown below.

image

## How has this issue affected you? What are you trying to accomplish?

I use light themes during the daytime, and it's difficult to read any text when I'm trying to add dependencies to my sandbox due to the poor contrast.

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      | latest (not sure where to check)
| Browser          | Chrome/80.0.3987.149
| Operating System | macOS 10.14.6

### PR DESCRIPTION
**Pull request by tanmoyopenroot**
## What kind of change does this PR introduce?

Bug fix #3796

## What is the current behavior?

On changing to light theme `Add dependencies` modal doesn't pick up the `light` theme

## What is the new behavior?

On Changing to `light` or `dark` theme appropriate colors are added to `Add dependencies` modal

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

The testing was done visually.

1. Select any `light` or `dark` theme from `Color Theme` under `Preferences`
2. Click on `Add dependency` in `Explorer tab`
3.  Appropriate colors are added to `Add dependencies` modal

Light Theme
Screenshot 2020-04-09 at 11 21 21 AM

Dark Theme
Screenshot 2020-04-09 at 11 21 00 AM

## Checklist

- [ ] Documentation
- [x] Testing 
- [x] Ready to be merged
      
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3865/commits/7f4687233124b52a4b2b7a804db98804debb9608 is at https://pr3865.build.csb.dev/s/new.

**Comment by CompuIves** [comment]
Thanks for this PR!!

File: packages/app/src/app/pages/common/Modals/index.tsx [hunk 1]
```diff
@@ -174,7 +174,7 @@ const Modals: FunctionComponent = () => {
   const modal = currentModal && modals[currentModal];
 
   return (
-    <ThemeProvider theme={localState.theme.vscodeTheme}>
+    <ThemeProvider theme={localState.theme}>
```
Comment (MichaelDeBoey) [inline packages/app/src/app/pages/common/Modals/index.tsx:177]: We need to be sure this doesn't break any modals other than the Search modal.

**Comment by siddharthkp** [comment]
> This looks very good visually, the only thing I notice is the border around the modal, would it be possible to remove that?

I think we have styles for that in the polyfill, can take it from here 👍

**Comment by MichaelDeBoey** [comment]
@siddharthkp The `ThemeProvider` has changed from `@codesandbox/components` to `styled-components`, so that's probably the cause of the border around _all_ modals

**Comment by tanmoyopenroot** [comment]
> Thanks for this PR!! This looks very good visually, the only thing I notice is the border around the modal, would it be possible to remove that?

I will try to fix it without breaking any other modals.

**Comment by tanmoyopenroot** [comment]
> @tanmoyopenroot This change also breaks opening the delete modal (both in a forked sandbox and on your dashboard).
> Probably other modals too, but haven't tested that.

I will fix it by using `ThemeProvider` from `@codesandbox/components`

**Comment by siddharthkp** [comment]
@tanmoyopenroot Would you be open to pair program on this bug? I can explain what we do in the `ThemeProvider` in `@codesandbox/components`

DM me on [twitter](https://twitter.com/siddharthkp) or email at sid @ codesandbox.io

**Comment by tanmoyopenroot** [comment]
> @tanmoyopenroot Would you be open to pair program on this bug? I can explain what we do in the `ThemeProvider` in `@codesandbox/components`

Sure, I will ping you.

**Comment by siddharthkp** [comment]
@tanmoyopenroot 👍 sent you an invite for slack

**Comment by tanmoyopenroot** [comment]
Dark Theme
Screenshot 2020-04-09 PM

Light Theme
Screenshot 2020-04-09 at 7 12 07 PM

File: packages/app/src/app/pages/Sandbox/SearchDependencies/dependencies.js [hunk 2]
```diff
@@ -32,24 +37,30 @@ export default createGlobalStyle`
   }
 
   .ais-Highlight-highlighted {
-       color: ${props => props.theme['button.hoverBackground']};
+    color: ${props => props.theme['button.hoverBackground']};
```
Comment (jyash97) [inline packages/app/src/app/pages/Sandbox/SearchDependencies/dependencies.js:40]: How about using `button.background` currently, the highlighted text isn't visible properly in light theme. And `button.background` goes in contrast with the background. Check the below screenshot:

Light Theme:
image

Dark theme:
image

Thoughts ?

Comment (tanmoyopenroot) [reply packages/app/src/app/pages/Sandbox/SearchDependencies/dependencies.js:40]: I and @siddharthkp had the same the discussion about the `highlighted` text issue.
And after using `button.background` the `highlighted` are visible now.

Comment (tanmoyopenroot) [reply packages/app/src/app/pages/Sandbox/SearchDependencies/dependencies.js:40]: I will make the changes and push it.

Comment (siddharthkp) [reply packages/app/src/app/pages/Sandbox/SearchDependencies/dependencies.js]: 👋 

short: this works, :shipit:! 

I'd recommend using <Text variant="muted"> and <Text> to create a difference between highlighted text and normal text.

It's hard to make this modal to match the redesign of the editor without redesigning the parts of the modal with `@codesandbox/components`.

That's not something that should block this improvement though! I'd recommend merging this PR and continue the work in another PR which can be longer term :)

**Pull request merged**